manim/animation/transform_matching_parts.py

```diff
@@ -12,6 +12,7 @@
 from ..mobject.opengl_mobject import OpenGLGroup, OpenGLMobject
 from ..mobject.types.opengl_vectorized_mobject import OpenGLVGroup, OpenGLVMobject
 from ..mobject.types.vectorized_mobject import VGroup, VMobject
+from ..scene.scene import Scene
```

Comment:
I think this import is only needed for the type, so can you make it so that it's only imported if typechecking? Such as is done [here](https://github.com/ManimCommunity/manim/blob/005adf48e599ac5cbaf98ad8ae379da2974646cd/manim/mobject/polyhedra.py#L12-L13) for example.